core-rules/src/main/java/org/hisp/dhis/android/rules/RuleVariableValue.java

```diff
@@ -0,0 +1,45 @@
+package org.hisp.dhis.android.rules;
+
+import com.google.auto.value.AutoValue;
+
+import org.hisp.dhis.android.rules.models.RuleValueType;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+@AutoValue
+abstract class RuleVariableValue {
+
+    @Nullable
+    abstract String value();
+
+    @Nonnull
+    abstract RuleValueType type();
+
+    @Nonnull
+    abstract List<String> candidates();
```

Comment:
No, because it is just another state of the same model by design. RuleVariableValue might not contain actual value, but can contain expected type (which comes from Variable). This way we can substitute default or "null" value for primitives which can be false, 0 or '' depending on source type.